pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -430,6 +452,18 @@ public PulsarClient getReplicationClient(String cluster) {
         }
 
         final CompletableFuture<Topic> topicFuture = new CompletableFuture<>();
+        final Semaphore topicLoadSemaphore = topicLoadRequestSemaphore.get();
+
+        if (!topicLoadSemaphore.tryAcquire()) {
+            if (failOnTooManyConcurrentTopics) {
+                throw new TooManyRequestsException("Too many concurrent topic load exception");
```

Comment:
the main idea behind failing immediately with exception so, while retrying create topic from [pendingTopicLoadingQueue](https://github.com/rdhabalia/pulsar/blob/406b0fc6b8efcc1451e163858a1e3f72860968fd/pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java#L1037) if it fails again here then we can put same topic-future again into the queue because it has original `future` which we had returned back to `ServerCnx` on very first time when `createPersistentTopic` request came.